src/components/admin/Login.jsx

```diff
@@ -1,73 +1,114 @@
 import BaseComponent from 'lib/BaseComponent';
 import React from 'react';
 import { browserHistory } from 'react-router';
-import { hash } from 'lib/utilities';
+import { googleClientID, isCI } from 'lib/utilities';
 import _ from 'lodash';
 
-import TextField from 'material-ui/TextField';
-import RaisedButton from 'material-ui/RaisedButton';
-
 const styles = {
-  buttons: {
-    marginTop: 12,
-    marginBottom: 24,
+  button: {
+    marginTop: 36,
   },
 };
 
+
 export default class Login extends BaseComponent {
   constructor(props) {
     super(props);
-    this.safeSetState({
-      passwordValue: '',
-    });
-    this.handlePasswordChange = this.handlePasswordChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+    this.state = {
+      GoogleAPIReady: false,
+    };
+    this.handleRedirect = this.handleRedirect.bind(this);
+    this.handleLoginCI = this.handleLoginCI.bind(this);
+    this.onSignInSuccess = this.onSignInSuccess.bind(this);
   }
 
-  handlePasswordChange(e) {
-    this.safeSetState({
-      passwordValue: e.target.value,
-    });
-  }
+  componentDidMount() {
+    super.componentDidMount();
+    const interval = setInterval(() => {
+      // GoogleAPILoaded is a global var, set to true when Google platform.js script loaded
+      if (window.THE_GAZELLE.googleAPILoaded) { // eslint-disable-line no-undef
+        this.renderButton();
+        window.gapi.load('auth2', () => {
+          this.safeSetState({
+            GoogleAPIReady: true,
+          });
 
-  handleSubmit(e) {
-    e.preventDefault();
-    const pass = e.target.password.value;
-    // eslint-disable-next-line max-len
-    if (hash(pass) !== 'eaafc81d7868e1c203ecc90f387acfa4c24d1027134b0bfda6fd7c536efc5d8dd5718609a407dbfcd41e747aec331153d47733153afb7c125c558acba3fb6bcd') {
-      window.alert('Incorrect password');
-      e.target.password.value = ''; // eslint-disable-line no-param-reassign
-    } else {
-      // Otherwise it is correct and we let the site redirect the client
-      let url = this.props.location.query.url || '';
-      if (url) {
-        _.forEach(this.props.location.query, (q, name) => {
-          if (name !== 'url') {
-            url = `${url}&${name}=${q}`;
+          const auth = window.gapi.auth2.getAuthInstance();
+          if (!auth || !auth.isSignedIn.get()) {
+            window.gapi.auth2.init({
+              client_id: googleClientID,
+            });
+          } else { // detects user has signed in
+            this.onSignInSuccess(auth.currentUser.get());
```

Comment:
Hmm it would be really great if we could have tests for these cases, but I guess we'll just skip them for now as it's cumbersome, and I'll create an issue with low priority about that we really should have this at some point